inc/sainat.h

```diff
@@ -0,0 +1,653 @@
+
+} sai_nat_zone_counter_attr_t;
+
+/**
+ * @brief Create and return a NAT zone counter object
+ *
+ * @param[out] nat_zone_counter_id NAT counter object
+ * @param[in] switch_id Switch object id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_create_nat_zone_counter_fn)(
+        _Out_ sai_object_id_t *nat_zone_counter_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Deletes a specified NAT zone_counter object.
+ *
+ * Deleting a NAT counter object does not delete reference to it.
+ *
+ * @param[in] nat_zone_counter_id NAT object to be removed.
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_remove_nat_zone_counter_fn)(
+        _In_ sai_object_id_t nat_zone_counter_id);
+
+/**
+ * @brief Set NAT zone counter attribute value(s).
+ *
+ * @param[in] nat_zone_counter_id NAT zone counter id
+ * @param[in] attr Attribute to set
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_set_nat_zone_counter_attribute_fn)(
+        _In_ sai_object_id_t nat_zone_counter_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get values for specified NAT zone counter attributes.
+ *
+ * @param[in] nat_zone_counter_id NAT counter zone object id
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_nat_zone_counter_attribute_fn)(
+        _In_ sai_object_id_t nat_zone_counter_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief NAT API Router entry methods table retrieved with sai_api_query()
+ */
+typedef struct _sai_nat_api_t
+{
+
+    /**
+     * @brief SAI NAT API set
+     */
+    sai_create_nat_entry_fn                   create_nat_entry;
+    sai_remove_nat_entry_fn                   remove_nat_entry;
+    sai_set_nat_entry_attribute_fn            set_nat_entry_attribute;
+    sai_get_nat_entry_attribute_fn            get_nat_entry_attribute;
+
+    sai_bulk_create_nat_entry_fn              create_nat_entries;
+    sai_bulk_remove_nat_entry_fn              remove_nat_entries;
+    sai_bulk_set_nat_entry_attribute_fn       set_nat_entries_attribute;
+    sai_bulk_get_nat_entry_attribute_fn       get_nat_entries_attribute;
+
+    sai_create_nat_zone_counter_fn            create_nat_zone_counter;
+    sai_remove_nat_zone_counter_fn            remove_nat_zone_counter;
+    sai_set_nat_zone_counter_attribute_fn     set_nat_zone_counter_attribute;
+    sai_get_nat_zone_counter_attribute_fn     get_nat_zone_counter_attribute;
+} sai_nat_api_t;
```

Comment:
How to create the zone itself?

Reply:
Zone is a namespace for operator and is configured as a string in schema. NAT application maps it to a uint_8 and provides this in SAI API.
Another good suggestion by Mike L. is to use say_nat_type_t and because of the impact of changes we are not considering this as well.